ion-schema/src/system.rs

```diff
@@ -862,27 +929,65 @@ impl Resolver {
         // get all isl type names that are defined within the schema
         // this will be used to resolve type references which might not have yet resolved while loading a type definition
         let isl_type_names: HashSet<&str> = HashSet::from_iter(
-            isl.types()
+            isl_types
                 .iter()
-                .filter(|t| matches!(t, IslType::Named(_)))
+                .filter(|t| matches!(t, IslTypeKind::Named(_)))
                 .map(|t| t.name().as_ref().unwrap().as_str()),
         );
 
         // Resolve all ISL types and constraints
         for isl_type in isl.types() {
             let pending_types = &mut PendingTypes::default();
 
-            match isl_type {
-                IslType::Named(named_isl_type) => {
+            match &isl_type.kind {
+                IslTypeKind::Named(named_isl_type) => {
+                    match isl_version {
+                        IonSchemaLanguageVersion::V1_0 => {
+                            // verify if there are any constraints with ISL 2.0 for this isl_type
+                            let has_isl_2_0_constraints =
+                                isl_type.constraints().iter().any(|c| match c.constraint {
+                                    IslConstraintImpl::Regex(_)
+                                    | IslConstraintImpl::Element(_)
+                                    | IslConstraintImpl::Annotations(_)
+                                    | IslConstraintImpl::Scale(_) => {
+                                        c.version == IonSchemaLanguageVersion::V2_0
+                                    }
+                                    _ => false,
+                                });
+
+                            if has_isl_2_0_constraints {
+                                return invalid_schema_error(format!("ISL type: {} contains constraints from ISL 2.0. Only use ISL 1.0 constraints for this method.", named_isl_type.name().as_ref().unwrap()));
+                            }
+                        }
+                        IonSchemaLanguageVersion::V2_0 => {
+                            // verify if there are any constraints with ISL 1.0 for this isl_type
+                            let has_isl_1_0_constraints =
+                                isl_type.constraints().iter().any(|c| match c.constraint {
+                                    IslConstraintImpl::Regex(_)
+                                    | IslConstraintImpl::Element(_)
+                                    | IslConstraintImpl::Annotations(_)
+                                    | IslConstraintImpl::Scale(_) => {
+                                        c.version == IonSchemaLanguageVersion::V2_0
```

Comment:
Should this be `IonSchemaLanguageVersion::V1_0`?

Reply:
That is correct. Happened while modularization the code. I will add some tests for this once we have some implementation of constraints from ISL 2.0. Currently those are all `todo!()`.